File: dll/argtypes/argtypes_defaults.py
import capi

dll = "dll/argtypes.dll"


@capi.func(dll)
def at_int(i):
    pass


@capi.func(dll)
def at_dbl(d):
    pass


@capi.func(dll)
def at_str(s):
    pass


if __name__ == "__main__":
    at_int(100)
    at_dbl(75.0123456789)
    at_str("this works")

File: dll/argtypes/dll/argtypes_defaults.c
#include <stdio.h>

void at_int(int i)
{
    printf("at_int: %d\n", i);
}

void at_dbl(double d)
{
    printf("at_dbl: %.10f\n", d);
}

void at_str(char* s)
{
    printf("at_str: ");
    printf(s);
    printf("\n");
}
